corehq/apps/app_manager/suite_xml/post_process/workflow.py

```diff
@@ -67,10 +67,10 @@ def _add_frames_to_entry(entry, frames):
         for frame in frames:
             entry.stack.add_frame(frame)
 
-    def get_frame_children(self, command, module, module_only=False, include_root_module=False):
+    def get_frame_children(self, module, form=None, include_root_module=False):
```

Comment:
The signature change in this method is the most substantial part of this PR's refactor.  I tried hard to ensure there were no behavior changes, and would appreciate reviewers validating that as well.

Reply:
One assumption I made (and checked) that might not be obvious is that `module` is always `form`'s module.

Reply:
I have mixed feelings about this signature change. The nice thing about passing `command` is that you know immediately that you could be dealing with either a form or a module, since a command can be either, so it acted like a `module_or_form` param, without the unpleasantness of having a param that can be two different types. Having a function call that contains both a form and that form's module bothers the DRY purist in me. But I do see your point about the awkward way `command` was used within the function.

With that complaining out of the way, my only constructive thought is to possibly have two methods, a `get_frame_children_for_module(self, module)` and a `get_frame_children_for_form(self, form, include_root_module=False)` which would both then call into this function. Up to you if that seems worthwhile.

Reply:
Yeah, it felt wrong to have all the callers generate an ID string from a form or case list and then pass that in here only to immediately regenerate the form and module.  Incidentally, the function ostensibly didn't support module commands, so callers that needed module info grabbed the 0th form from the module and passed in that for the command, with `module_only=True`.  I'm not bothered by passing in `form` and `module`, and I think it's less awkward than a single `module_or_form` arg.  I do agree that two separate methods would provide a cleaner-still interface.  It sounds kind of hard to disentangle the two though, so I think you'd either end up with the same thing, but as a private method, or something similarly complex.  I'll punt on that, but wouldn't be offended if you decided to make that change as you work with this code.

As an aside, I think `include_root_module` is a better default behavior (in principle anyways, weird current behavior notwithstanding).  I'd be interested to see whether all of the callers should be taking advantage of it.  I know at least the EoF stuff and now session endpoints reimplement the same principle in a slightly different way.

Reply:
> I think you'd either end up with the same thing, but as a private method, or something similarly complex.

yeah, I agree this is likely